seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/util/ClickhouseProxy.java

```diff
@@ -274,6 +274,58 @@ public ClickhouseTable getClickhouseTable(String database, String table) {
         }
     }
 
+    /**
+     * Get ClickHouse table info for shard node.
+     *
+     * @param request
+     * @param database
+     * @param table
+     * @return
+     */
+    public ClickhouseTable getClickhouseTableForShardNode(
+            ClickHouseRequest<?> request, String database, String table) {
```

Comment:
why not just use request in https://github.com/apache/seatunnel/pull/8296/files#diff-ec9cf2e8c4d74f805e8558e079dcd9941afed65045288075c9766fe07b487ab7L57?

Reply:
This depends on the implementation of clickhouse client. ClickhouseRequest uses http connection and will randomly access a copy of the corresponding node. This change is similar to the implementation in ClickhouseFileSinkAggCommitter.
```
    private void attachFileToClickhouse(Shard shard, List<String> clickhouseLocalFiles)
            throws ClickHouseException {
        ClickHouseRequest<?> request = getProxy().getClickhouseConnection(shard);
        for (String clickhouseLocalFile : clickhouseLocalFiles) {
            ClickHouseResponse response =
                    request.query(
                                    String.format(
                                            "ALTER TABLE %s ATTACH PART '%s'",
                                            clickhouseTable.getLocalTableName(),
                                            clickhouseLocalFile.substring(
                                                    clickhouseLocalFile.lastIndexOf("/") + 1)))
                            .executeAndWait();
            response.close();
        }
    }
```